### ISSUE TITLE
Error message when trying to launch xfce4-terminal with sudo in spacefm toolbar command

### ISSUE THREAD

**Issue by HerborenInVlammen** [IgnorantGuru/spacefm#551]
Trying to launch xfce4-terminal with sudo from a spacefm command (toolbar, menu etc.) gives following error no matter what is done even though the terminal is launched:

Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined

Launching other terminals with sudo does not give this error.  But I have xfce4-terminal set as the default terminal in spacefm.  Is there any way around this?

**Comment by IgnorantGuru** [comment]
Not sure why you closed this but it looks more like a warning message.  Also note that SpaceFM passes xfce4-terminal the --disable-server option (so it runs one terminal per instance).  Generally speaking, each terminal has its own options and SpaceFM does some different things for each, and sometimes they need to be tweaked when things are changed in option usage (not only do most terminals differ in options, but they change them with backwards compatibility as well).  If you find another option that should be passed you can detail it here or add it after the terminal name in Prefs.

**Comment by HerborenInVlammen** [comment]
Hi:  I closed it because I discovered it has nothing to do with
Spacefm.  It is a known XFCE issue and is 100% reproducible from outside
Spacefm.  Sometimes I get tunnel vision trying to solve issues that I
miss the forest for the trees.  The first thing I should have done
trying to solve it should have been to try it outside of spacefm.

It happens when you use sudo to launch an instance of xfce4-terminal
from within a xfce4-terminal (ya, I know there is absolutely no need to
do so within Spacefm, but I was messing around... lol).

Anyway, if one uses gksudo, the problem is no more.

P.S., even passing the appropriate SESSION_MANAGER variable does not fix
it.
